packages/core-data/src/types/attachment.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Internal dependencies
+ */
+import {
+	Context,
+	ContextualField,
+	MediaType,
+	PostStatus,
+	RenderedText,
+	OmitNevers,
+} from './helpers';
+
+export interface ExtensibleAttachment< C extends Context > {
+	/**
+	 * The date the post was published, in the site's timezone.
+	 */
+	date: string | null;
+	/**
+	 * The date the post was published, as GMT.
+	 */
+	date_gmt: ContextualField< string | null, 'view' | 'edit', C >;
+	/**
+	 * The globally unique identifier for the post.
+	 */
+	guid: ContextualField< RenderedText< C >, 'view' | 'edit', C >;
+	/**
+	 * Unique identifier for the post.
+	 */
+	id: number;
+	/**
+	 * URL to the post.
+	 */
+	link: string;
+	/**
+	 * The date the post was last modified, in the site's timezone.
+	 */
+	modified: ContextualField< string, 'view' | 'edit', C >;
+	/**
+	 * The date the post was last modified, as GMT.
+	 */
+	modified_gmt: ContextualField< string, 'view' | 'edit', C >;
+	/**
+	 * An alphanumeric identifier for the post unique to its type.
+	 */
+	slug: string;
+	/**
+	 * A named status for the post.
+	 */
+	status: ContextualField< PostStatus, 'view' | 'edit', C >;
+	/**
+	 * Type of post.
+	 */
+	type: string;
+	/**
+	 * Permalink template for the post.
+	 */
+	permalink_template: ContextualField< string, 'edit', C >;
+	/**
+	 * Slug automatically generated from the post title.
+	 */
+	generated_slug: ContextualField< string, 'edit', C >;
+	/**
+	 * The title for the post.
+	 */
+	title: RenderedText< C >;
+	/**
+	 * The ID for the author of the post.
+	 */
+	author: number;
+	/**
+	 * Whether or not comments are open on the post.
+	 */
+	comment_status: ContextualField< 'open' | 'closed', 'view' | 'edit', C >;
```

Comment:
should we use `CommentingStatus` here?

it's fine not to, but if we are going to have the type on `helpers` we might as well use it everywhere.

Reply:
I missed these two, thanks!